app/models/application_choice.rb

```diff
@@ -2,7 +2,9 @@ class ApplicationChoice < ApplicationRecord
   before_create :set_id
   before_create :set_initial_status
   belongs_to :application_form, touch: true
-  belongs_to :course
+  belongs_to :course_option
```

Comment:
I am not sure about the name of the model `course_option`,
when we use `application_choise.course_option`, we actually get the set of options for the `application_choise`, and not just one option as the name suggests. 
This is a common issue with active record, since we are "forced" to use a singular name.
We may want to use some of this alternative:
`course_option_set`,  `course_option_group`, `course_option_collection` , ...